packages/clerk-js/src/core/clerk.ts

```diff
@@ -2200,6 +2169,18 @@ export class Clerk implements ClerkInterface {
       }
     }
 
+    const setActiveNavigate = async ({ session, redirectUrl }: { session: SessionResource; redirectUrl: string }) => {
+      if (!session.currentTask) {
+        await this.navigate(redirectUrl);
+        return;
+      }
+
+      await navigateIfTaskExists(session, {
+        baseUrl: displayConfig.signInUrl,
+        navigate: this.navigate,
```

Comment:
I've tried sending the `customNavigate` here to avoid unmounting the component route when switching from hash routing to path routing, but it handles an absolute path as relative path and breaks the navigation. 

I think our current approach is to send a relative path via prop to `SignInSsoCallback/SignUpSsoCallback` to pass to the React context `navigate` reference. 

We could pass `taskUrl` as `../tasks`, but it'd lead to a double navigation (although with path routing) -> `/sign-in/sso-callback` -> `/sign-in/tasks` -> `/sign-in/choose-organization`. It's fast tho, so could be better than unmounting the component.

@panteliselef @nikosdouvlis thoughts?